test/e2e/specs/interactivity/router-script-modules.spec.ts

```diff
@@ -55,61 +55,112 @@ test.describe( 'Router script modules', () => {
 		await utils.deleteAllPosts();
 	} );
 
-	test( 'should handle modules from new blocks', async ( { page } ) => {
-		const requestedModules = [];
-
-		await page.route( '**/*.js*', async ( route ) => {
-			requestedModules.push( route.request().url() );
-			await route.continue();
+	for ( const [ testId, buttonId, expectedValues ] of [
+		[
+			'static modules from new blocks',
+			'static',
+			{ alpha: 'alpha-1', bravo: 'bravo-1', charlie: 'charlie-1' },
+		],
+		[
+			'dynamic modules from new blocks',
+			'dynamic',
+			{ alpha: 'alpha-2', bravo: 'bravo-2', charlie: 'charlie-2' },
+		],
+		[
+			'static modules from initial page',
+			'initial-static',
+			{ alpha: 'initial-1', bravo: 'initial-1', charlie: 'initial-1' },
+		],
+		[
+			'dynamic modules from initial page',
+			'initial-dynamic',
+			{ alpha: 'initial-2', bravo: 'initial-2', charlie: 'initial-2' },
+		],
+	] as const ) {
+		test( `should handle ${ testId }`, async ( { page } ) => {
+			const requestedModules = [];
+
+			await page.route( '**/*.js*', async ( route ) => {
+				requestedModules.push( route.request().url() );
+				await route.continue();
+			} );
+
+			const csn = page.getByTestId( 'client-side navigation' );
+			const alpha = page.getByTestId( 'alpha-block' );
+			const bravo = page.getByTestId( 'bravo-block' );
+			const charlie = page.getByTestId( 'charlie-block' );
+
+			await page.getByTestId( 'link alpha' ).click();
+
+			// This element disappears when a navigation starts.
+			// It should be visible again after a successful navigation.
+			await expect( csn ).toBeHidden();
+			await expect( csn ).toBeVisible();
+
+			// Check the page title to ensure navigation was successful.
+			await expect( page ).toHaveTitle( 'alpha – gutenberg' );
```

Comment:
Shouldn't it be like this?

```suggestion
			// This element disappears when a navigation starts.
			await expect( csn ).toBeHidden();

			// Check the page title to ensure navigation was successful.
			await expect( page ).toHaveTitle( 'alpha – gutenberg' );

			// It should be visible again after a successful navigation.
			await expect( csn ).toBeVisible();
```

Reply:
I updated it in https://github.com/WordPress/gutenberg/pull/70489/commits/35bfedf26c71c9882ba2290eaf3f18cb7a0c5060.